test/golden_output_test.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Artman golden tests.
+
+It runs GAPIC generation against the golden input data, can then compare the
+output layout with the expected output. This helps to capture unexpected GAPIC
+output layout change caused by either by artman, toolkit or other runtime
+dependency.
+
+Note: a set of static input files (e.g. google/api/http.proto) are being used
+for the golden test, so it is not going to capture upstream changes to such
+input.
+"""
+
+import os
+import subprocess
+
+
+def run_golden_test():
+    golden_dir = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), 'testdata', 'golden')
+    expected = []
+    with open(os.path.join(golden_dir, 'output', 'expected_output.txt')) as f:
+        expected = f.read().splitlines()
+
+    for lang in ['python', 'java', 'ruby', 'nodejs', 'php', 'go', 'csharp']:
+        generate_gapic_library(golden_dir, lang)
+
+    output_dir = '/output'
+    actual = []
+    for root, subdirs, files in os.walk(output_dir):
+        for f in files:
+            actual.append(os.path.join(root, f)[len(output_dir):])
+
+    expected.sort()
+    actual.sort()
+    if expected != actual:
```

Comment:
This is actually the opposite of what I was asking for. I wanted there to be a canonical ordering so that the baseline file will be stable, instead of permitting arbitrary reorderings of the lines of the file between commits.